sfxclient/sfxclient.go

```diff
@@ -133,32 +143,64 @@ func (c *callbackPair) getDatapoints(now time.Time, sendZeroTime bool) []*datapo
 	return ret
 }
 
+func (c *callbackPair) getDatapointsWithDebug(parentSpan opentracing.Span, now time.Time, sendZeroTime bool) []*datapoint.Datapoint {
+	var (
+		buf bytes.Buffer
+		ret = make([]*datapoint.Datapoint, 0, c.expectedSize)
+	)
+
+	for callback := range c.callbacks {
+		buf.WriteString(reflect.TypeOf(callback).String())
+		buf.WriteString(spanDatapointSuffix)
+		span := opentracing.GlobalTracer().StartSpan(buf.String(), opentracing.ChildOf(parentSpan.Context()))
+		startTime := time.Now()
+		ret = append(ret, callback.Datapoints()...)
+		span.SetTag("time-taken", time.Since(startTime).String())
+		span.Finish()
+		buf.Reset()
+	}
+	for _, dp := range ret {
```

Comment:
this section is a perfect copy of the bottom half of getDatapoints, refactor?

Reply:
yes for sure